tests/integration/targets/api/tasks/main.yml

```diff
@@ -0,0 +1,221 @@
+      servicenow.itsm.api: *api-delete-data
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+
+    - name: Verify that record was deleted
+      servicenow.itsm.api_info:
+        resource: incident
+        sys_id: "{{ first_incident.record.sys_id }}"
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.record == []
+
+    # Despite the module not being idempotent, there is no resource with the mentioned sys_id,
+    # so this element will not get deleted once again.
+    - name: Delete a problem (idempotency)
+      servicenow.itsm.api: *api-delete-data
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is not changed
+
+    - name: Test bad resource name
+      servicenow.itsm.api_info:
+        resource: bad_table_name
+      ignore_errors: true
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is failed
+          - "'Invalid table bad_table_name' in result.msg"
+
+    - name: Test invalid column name - API ignores column names that are not present in the resource
+      servicenow.itsm.api_info:
+        resource: incident
+        sysparm_query: bad_column=bad_column_value1^ORbad_column=bad_column_value2 # 'bad_column' is not present in incident's columns
+      register: result_bad_query
+    - name: Retrieve all incidents so they'll be compared to the API call with columns ignored
+      servicenow.itsm.api_info:
+        resource: incident
+      register: result_no_query
+    - ansible.builtin.assert:
+        that:
+          - result_bad_query.record | length == result_no_query.record | length
+
+    # Using api module's 'column' field
+    - name: Create a resource in table incident with attributes, specified in template file
+      servicenow.itsm.api:
+        resource: incident
+        action: post
+        template: "test_template_post.j2"
+      vars: &template-vars
+        short_desc: "short"
+        long_desc: "looooooooong description"
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+          - result.record.short_description == 'short'
+          - result.record.description == 'short looooooooong description'
+
+    - name: Update a resource in table incident with attributes, specified in template file
+      servicenow.itsm.api:
+        resource: incident
+        action: patch
+        sys_id: "{{ result.record.sys_id }}"
+        template: "test_template_patch.j2"
+      vars: *template-vars
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+          - "result.record.short_description == 'Changed template - short'"
+          - "result.record.description == 'Desciption is changed also - short looooooooong description'"
+
+    # - name: print info from previous task
+    #   ansible.builtin.debug:
+    #     var: result
```

Comment:
```suggestion
```
Remove debugging code if not required.

Reply:
Ugh, missed that one, sorry. Resolved with b01b4f77282f50c9e544f68b7b90586a56c29944.